expertsystem/ui/__init__.py

```diff
@@ -239,84 +339,6 @@ def _determine_node_settings(self, graphs):
             graph_node_setting_pairs.append((instance, node_settings))
         return graph_node_setting_pairs
 
-    def find_solutions(
```

Comment:
Why did you move this up?  My logic here was to keep functions called by other functions close to each other, so that the user does not need to scroll around too much

Reply:
To keep the public functions together (1c7216aec94b0370b8f58ae7113fe455debc6394). It's more for the developer, in that sense it's indeed better to keep it on the same place

Reply:
18e8684d48fcb993e71451578fd688a335873009

Reply:
Ok. Although I would be fine with such a change in general. We should have some kind of decision on this and then write this down in and adr or the docs. I want to avoid such changes going back and forth and having no consistent manner.

Reply:
It's hard to pin down. Sometimes you need to have the class defined first. Maybe one prefers public methods first. And then in some cases having the definition coming right after the call is easier for readability.

Reply:
I just followed the recommendation from clean code i believe (which made sense to me). I guess nowadays with strong IDEs this not as crucial anymore, but i think its still kind of nice to have thinks stuck together. So I would actually vote for how it is now. But I can be convinced otherwise :)

Reply:
Yep it's from Clean Code ^^